compiler/rustc_mir_build/src/thir/pattern/const_to_pat.rs

```diff
@@ -413,7 +441,7 @@ impl<'a, 'tcx> ConstToPat<'a, 'tcx> {
                                 suffix: vec![],
                             }),
                             span,
-                            ty: pointee_ty,
+                            ty: tcx.mk_slice(elem_ty),
```

Comment:
If I understand correctly, here you're making `b"foo"` literals always have type `&[u8]`? Wouldn't that break in a case where they really are supposed to be `&[u8;N]`? Though knowing how exhaustiveness handles slices it might work by accident.

Reply:
There's a match arm for `ty::Array` above, which handles all situations where we expect an array. I had to invert the condition, because I can't put conditions on or patterns, so I can't do `ty::Array(..) if something | ty::Slice(_)`

Reply:
Ah, got it